src/controller/python/OpCredsBinding.cpp

```diff
@@ -687,4 +748,29 @@ PyChipError pychip_GetCompletionError()
     return ToPyChipError(sTestCommissioner.GetCompletionError());
 }
 
+extern "C" {
+// Define a global callback to set the RCAC data callback
+using PythonRCACCallback                      = void (*)(const uint8_t *, size_t);
+static PythonRCACCallback gPythonRCACCallback = nullptr;
+
+// Function to set the Python callback
+void pychip_SetCommissioningRCACCallback(PythonRCACCallback callback)
+{
+    ChipLogProgress(Controller, "Attempting to set Python RCAC callback in C++");
+    gPythonRCACCallback = callback;
+
+    // Set C++ TestCommissioner callback to notify Python when RCAC data is ready
+    sTestCommissioner.SetOnRCACReadyCallback([](const std::vector<uint8_t> & rcacData) {
+        if (gPythonRCACCallback)
+        {
+            ChipLogProgress(Controller, "RCAC callback in C++ set");
+            gPythonRCACCallback(rcacData.data(), rcacData.size());
+        }
+        else
+        {
+            ChipLogError(Controller, "Python RCAC callback is not set.");
```

Comment:
You can omit this else - gPythonRCACCallback was set in this same function. If it was set to nullptr, it was intentional.

Reply:
Thank you, I will remove this from the code!